src/components/shared/Navigation.tsx

```diff
@@ -0,0 +1,180 @@
+import "es6-symbol/implement";
+import * as React from "react";
+import { Text } from "react-native";
+import Icon from "react-native-vector-icons/MaterialCommunityIcons";
+import Ionicons from "react-native-vector-icons/Ionicons";
+
+import { Home } from "../pages/Home";
+import { LogIn } from "../pages/LogIn";
+import { OnboardingCamera } from "../pages/OnboardingCamera";
+import { Onboarding } from "../pages/Onboarding";
+import { MyProfile } from "../pages/MyProfile";
+import { VideoPreview } from "../pages/VideoPreview";
+import { getMembersByIds } from "../../../src/store/selectors/members";
+import { RahaState } from "../../../src/store";
+import { createMaterialBottomTabNavigator } from "react-navigation-material-bottom-tabs";
+import { createStackNavigator } from "react-navigation";
+import { connect, MapDispatchToProps, MapStateToProps } from "react-redux";
+
+export enum RouteName {
+  Home = "Home",
+  Onboarding = "Onboarding",
+  OnboardingCamera = "OnboardingCamera",
+  LogIn = "LogIn",
+  MyProfile = "MyProfile",
+  VideoPreview = "VideoPreview",
+  Search = "Search",
+  Invite = "Invite",
+  Give = "Give"
+}
+
+const SignedIn = createMaterialBottomTabNavigator(
+  {
+    Home: {
+      screen: Home
+    },
+    Search: {
+      screen: OnboardingCamera
+    },
+    Give: {
+      screen: OnboardingCamera
+    },
+    Invite: {
+      screen: OnboardingCamera
+    },
+    MyProfile: {
+      screen: MyProfile
+    }
+  } as { [key in RouteName]: any }, // TODO: once react-nav types in, edit
+  {
+    initialRouteName: RouteName.MyProfile,
+    labeled: false,
+    navigationOptions: ({ navigation }: any) => ({
+      tabBarIcon: ({ focused }: any) => {
+        const { routeName } = navigation.state;
+        let iconName = null;
+        let IconType = Icon;
+        switch (routeName) {
+          case RouteName.MyProfile:
+            iconName = "account";
+            break;
+          case RouteName.Give:
+            iconName = "star";
+            break;
+          case RouteName.Home:
+            iconName = "home";
+            break;
+          case RouteName.OnboardingCamera:
+            iconName = "account-multiple-plus";
+            break;
+          case RouteName.Invite:
+            iconName = "account-multiple-plus";
+            break;
+          case RouteName.Search:
+            iconName = "ios-search";
+            IconType = Ionicons;
+            break;
+          default:
+            iconName = "bad";
```

Comment:
What does `bad` mean? If this is supposed to never happen, I say put a `console.error` call, a `TODO` comment to remind us to log it properly in the future then return nothing (like an empty `<View />`)